hypertrace-trace-enricher/enriched-span-constants/src/main/java/org/hypertrace/traceenricher/enrichedspan/constants/utils/EnrichedSpanUtils.java

```diff
@@ -301,13 +303,25 @@ public static String getUserAgent(Event event) {
   }
 
   public static Optional<String> getFullHttpUrl(Event event) {
-    if (event.getHttp() != null && event.getHttp().getRequest() != null) {
-      return Optional.ofNullable(event.getHttp().getRequest().getUrl());
+    if (event.getHttp() != null
+        && event.getHttp().getRequest() != null
+        && event.getHttp().getRequest().getUrl() != null
+        && isAbsoluteUrl(event.getHttp().getRequest().getUrl())) {
```

Comment:
when the URL isn't full URL, do you want to rather check if you can construct full URL back from scheme, authority, path and query params? That's what I had in mind.

Reply:
Oh.. if you're making sure this URL is always full URL, we don't even need this method I think.

Reply:
We would need it. 
Earlier, event.getHttp().getRequest().getUrl() was null in case of relative url. Now since it is being populated in a best effort basis, so it maybe full url or not.  
However the consumer of this method EnrichedSpanUtils#getFullHttpUrl would except a full url to be returned or null. 
To ensure that behaviour a check has been added to this method. 

However I think we can go with this pr first https://github.com/hypertrace/hypertrace-ingester/pull/104 for now. It is a subset of the changes made in this pr.